sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_sender.py

```diff
@@ -368,48 +378,44 @@ def send_messages(self, message, **kwargs):
-            message = transform_messages_to_sendable_if_needed(message)
-            try:
-                for each_message in iter(message):  # type: ignore # Ignore type (and below) as it will except if wrong.
-                    add_link_to_send(each_message, send_span)
-                batch = self.create_message_batch()
-                batch._from_list(message, send_span)  # type: ignore # pylint: disable=protected-access
-                message = batch
-            except TypeError:  # Message was not a list or generator. Do needed tracing.
-                if isinstance(message, ServiceBusMessageBatch):
-                    for (
-                        batch_message
-                    ) in message.message._body_gen:  # pylint: disable=protected-access
-                        add_link_to_send(batch_message, send_span)
-                elif isinstance(message, ServiceBusMessage):
-                    trace_message(message, send_span)
-                    add_link_to_send(message, send_span)
+            if isinstance(message, ServiceBusMessageBatch):
+                for (
+                    batch_message
+                ) in message.message._body_gen:  # pylint: disable=protected-access
+                    add_link_to_send(batch_message, send_span)
+                obj_message = message  # type: MessageObjTypes
+            else:
+                obj_message = create_messages_from_dicts_if_needed(message, ServiceBusMessage)
+                # Ensure message is sendable (not a ReceivedMessage), and if needed (a list) is batched. Adds tracing.
+                obj_message = transform_messages_to_sendable_if_needed(obj_message)
+                try:
+                    # Ignore type (and below) as it will except if wrong.
+                    for each_message in iter(obj_message):  # type: ignore
+                        add_link_to_send(each_message, send_span)
+                    batch = self.create_message_batch()
+                    batch._from_list(obj_message, send_span)  # type: ignore # pylint: disable=protected-access
+                    obj_message = batch
+                except TypeError:  # Message was not a list or generator. Do needed tracing.
+                    trace_message(cast(ServiceBusMessage, obj_message), send_span)
+                    add_link_to_send(obj_message, send_span)
 
             if (
-                isinstance(message, ServiceBusMessageBatch) and len(message) == 0
+                isinstance(obj_message, ServiceBusMessageBatch) and len(obj_message) == 0
             ):  # pylint: disable=len-as-condition
                 return  # Short circuit noop if an empty list or batch is provided.
-            if not isinstance(message, (ServiceBusMessageBatch, ServiceBusMessage)):
-                raise TypeError(
-                    "Can only send azure.servicebus.<ServiceBusMessageBatch, ServiceBusMessage> "
-                    "or lists of ServiceBusMessage."
-                )
```

Comment:
would this removed part of instance type checking be covered by the `_single_message_from_dict` method?

Reply:
Yes that was my intention - it seems we perform the same type checks a number of times, so I tried to slim it down a bit.

Reply:
yea, I re-read the code, should be covered :)